tests/check_framework/test_database.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+
+from django.core import checks
+from django.core.checks.registry import CheckRegistry
+from django.db import connection
+from django.test import TestCase, mock
+
+
+class DatabaseCheckTests(TestCase):
+    @property
+    def func(self):
+        from django.core.checks.database import check_database_connections
+        return check_database_connections
+
+    def test_database_checks_not_run_by_default(self):
+        """
+        `database` checks are only run when their tag is specified.
+        """
+        def f1(**kwargs):
+            return [5]
+
+        registry = CheckRegistry()
+        registry.register("database")(f1)
+        errors = registry.run_checks()
+        self.assertEqual(errors, [])
+
+        errors2 = registry.run_checks(tags=["database"])
+        self.assertEqual(errors2, [5])
+
+    def test_database_checks_called(self):
+        with mock.patch('django.db.backends.base.validation.BaseDatabaseValidation.check') as mocked_check:
+            checks.run_checks(tags=["database"])
+            self.assertTrue(mocked_check.called)
+
+    @unittest.skipUnless(connection.vendor == 'mysql', "Test only for MySQL")
+    def test_mysql_strict_mode(self):
+        good_sql_modes = [
+            'STRICT_TRANS_TABLES,STRICT_ALL_TABLES',
+            'STRICT_TRANS_TABLES',
+            'STRICT_ALL_TABLES',
+            'TRADITIONAL',
```

Comment:
There's no need to accept "Traditional". When you *set* traditional, MySQL sets multiple modes, including `STRICT_TRANS_TABLES`. If you then ask it back what `sql_mode` is, you get them all back:

```
adamj@localhost [1]> set sql_mode = 'traditional';
Query OK, 0 rows affected (0.01 sec)

adamj@localhost [2]> select @@sql_mode;
+------------------------------------------------------------------------------------------------------------------------------------------------------+
| @@sql_mode                                                                                                                                           |
+------------------------------------------------------------------------------------------------------------------------------------------------------+
| STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION |
+------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```